src/ripple/peerfinder/impl/Logic.h

```diff
@@ -392,6 +392,11 @@ class Logic
         if (keys_.find (key) != keys_.end())
             return Result::duplicate;
 
+        // If the peer belongs to a cluster, update the slot to reflect that.
+        counts_.remove (*slot);
+        slot->cluster (cluster);
+        counts_.add (*slot);
```

Comment:
If the slot cannot activate we've upped `counts_` and return, which differs from the prior behavior.

Albeit ugly, we could preserve the prior behavior with...
```
        // If the peer belongs to a cluster, update the slot to reflect that.
        auto const saved {slot->cluster()};
        slot->cluster(cluster);

        // See if we have an open space for this slot
        if (! counts_.can_activate (*slot))
        {
            if (! slot->inbound())
                bootcache_.on_success (
                    slot->remote_endpoint());
            return Result::full;
        }

        slot->cluster(saved);
        counts_.remove(*slot);
        slot->cluster(cluster);
        counts_.add(*slot);
```

Reply:
Would you need to reset `slot->cluster(saved)` before the `return Result::full` too?

Reply:
@thejohnfreeman Most likely, good catch.

Reply:
The `counts_` should be properly decremented when the slot is released because the connection will be dropped. That should invoke [`Counts::remove`](https://github.com/ripple/rippled/blob/381a1b948b06d9526cc73f14cfc69635fabf8605/src/ripple/peerfinder/impl/Counts.h#L65) which will then call [`Counts::adjust`](https://github.com/ripple/rippled/blob/381a1b948b06d9526cc73f14cfc69635fabf8605/src/ripple/peerfinder/impl/Counts.h#L254).